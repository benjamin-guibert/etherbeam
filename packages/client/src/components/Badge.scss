@use '../styles/resources/sizes';

.my-badge {
  border-radius: sizes.$border-radius-s;
  padding: 0px sizes.$space-xs;
}
