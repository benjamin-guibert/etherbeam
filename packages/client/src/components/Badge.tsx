import React, { ReactElement, ReactNode } from 'react'
import './Badge.scss'

type BadgeTypeProp = 'neutral' | 'primary' | 'secondary' | 'success' | 'danger'

interface BadgeProps {
  type?: BadgeTypeProp
  className?: string
  children?: ReactNode
}

const Badge = ({ type = 'neutral', className, children }: BadgeProps): ReactElement => {
  return <span className={[`my-badge my-color-${type}`, className].join(' ')}>{children}</span>
}

export default Badge
