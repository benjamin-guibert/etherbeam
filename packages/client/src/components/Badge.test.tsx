import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Badge from './Badge'

describe('<Badge />', () => {
  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<Badge className="my-class">Content.</Badge>)
    })

    it('should render', () => {
      const span = component.find('span').at(0)
      expect(span.hasClass('my-badge')).toBeTruthy()
      expect(span.hasClass('my-color-neutral')).toBeTruthy()
      expect(span.hasClass('my-class')).toBeTruthy()
      expect(span.prop('children')).toBe('Content.')
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Type', () => {
    beforeAll(() => {
      component = shallow(<Badge type="primary">Content.</Badge>)
    })

    it('should render', () => {
      const span = component.find('span').at(0)
      expect(span.hasClass('my-badge')).toBeTruthy()
      expect(span.hasClass('my-color-primary')).toBeTruthy()
      expect(span.prop('children')).toBe('Content.')
    })
  })
})
