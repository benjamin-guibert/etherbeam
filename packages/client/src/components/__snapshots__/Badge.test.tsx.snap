// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> Default should match snapshot 1`] = `
<span
  class="my-badge my-color-neutral my-class"
>
  Content.
</span>
`;
